src/components/Match/MatchStory.jsx

```diff
@@ -0,0 +1,545 @@
+/* global API_HOST */
+import React from 'react';
+import strings from 'lang';
+import {
+  formatSeconds,
+  jsonFn,
+} from 'utility';
+import { IconRadiant, IconDire } from 'components/Icons';
+import heroes from 'dotaconstants/build/heroes.json';
+import items from 'dotaconstants/build/items.json';
+import styles from './Match.css';
+
+const heroesArr = jsonFn(heroes);
+const radiantColor = '#66bb6a';
+const direColor = '#ff4c4c';
+
+// can be used in conjunction with is_radiant
+const TEAM = {
+  radiant: true,
+  dire: false,
+};
+
+const GoldSpan = amount => (
+  <span key={`gold_${amount}`} className={styles.storySpan}>
+    <font color={styles.golden}>{amount} </font>
+    <img
+      role="presentation"
+      src={`${API_HOST}/apps/dota2/images/tooltips/gold.png`}
+    />
+  </span>
+);
+
+const TeamSpan = isRadiant => (
+  <span key={`team_${isRadiant ? 'radiant' : 'dire'}`} style={{ color: (isRadiant ? radiantColor : direColor) }} className={styles.storySpan}>
+    {isRadiant ? <IconRadiant className={styles.iconRadiant} /> : <IconDire className={styles.iconDire} />}
+    {isRadiant ? strings.general_radiant : strings.general_dire}
+  </span>
+);
+
+// Modified version of PlayerThumb
+const PlayerSpan = (player) => {
+  if (!player || !heroes[player.hero_id]) {
+    return 'Unrecognized Hero';
+  }
+  return (<span key={`player_${player.player_slot}`} style={{ color: (player.isRadiant ? radiantColor : direColor) }} className={styles.storySpan}>
+    <img
+      src={heroes[player.hero_id]
+        ? `${API_HOST}${heroes[player.hero_id].icon}`
+        : '/assets/images/blank-1x1.gif'
+      }
+      role="presentation"
+    />
+    {heroes[player.hero_id] ? heroes[player.hero_id].localized_name : 'Unrecognized Hero'}
+  </span>);
+};
+
+// Modified version of PlayerThumb
+const ItemSpan = item => (
+  <span key={`item_${item}`} className={styles.storySpan}>
+    <img
+      src={items[item]
+        ? `${API_HOST}${items[item].img}`
+        : '/assets/images/blank-1x1.gif'
+      }
+      role="presentation"
+    />
+    {items[item].dname}
+  </span>
+);
+
+const capitalizeFirst = (list) => {
+  if (typeof list[0] === 'string' || list[0] instanceof String) {
```

Comment:
instead of mutating list it's probably better to return a new string/Array with the desired mutation.  More functional programming usually discourages functions with side effects.